task_sdk/src/airflow/sdk/definitions/connection.py

```diff
@@ -50,3 +55,32 @@ class Connection:
     def get_uri(self): ...
 
     def get_hook(self): ...
+
+    def _get_extra_dejson(self, nested: bool = False) -> dict:
+        """
+        Deserialize extra property to JSON.
+
+        :param nested: Determines whether nested structures are also deserialized into JSON (default False).
+        """
+        extra = {}
+        log = structlog.get_logger(logger_name="task")
+
+        if self.extra:
+            try:
+                if nested:
+                    for key, value in json.loads(self.extra).items():
+                        extra[key] = value
+                        if isinstance(value, str):
+                            with suppress(JSONDecodeError):
+                                extra[key] = json.loads(value)
+                else:
+                    extra = json.loads(self.extra)
+            except JSONDecodeError:
+                log.error("Failed to deserialize extra property `extra`, returning empty dictionary")
+        # Mask sensitive stuff here
```

Comment:
Do we really need to keep this? What plans to migrate masking, will it be in the server or task sdk?

Reply:
I just realized I had added some part of `extra_dejson` yesterday night before creating that issue:

https://github.com/apache/airflow/pull/45444

Reply:
Should we mask it here - Probably not?

Reply:
Ah didn't see that. Let me rebase and adjust it out here

Reply:
About masking - Let me still leave a TODO, but I think the server should handle it?

Reply:
Added in 49950cbcbec7764477cddcf4d86351ee6a0ba155